### PR TITLE
TGB - "Rising Star" Ultra Low Pop Map

### PR DESCRIPTION
**Pull request by XSlayer300**
## About The Pull Request
A RP map designed for RP, since we don't do anything at 0 - 3 players short, other than Crash. It's based on the Marine Ball that didn't ran a few months back, due to LaKiller's leave. I also added some rooms that were not added from the map, such as a Killhouse and a a field where you can make a FOB in.

I'm currently planning features such as an auto-end vote once they're 5 players or more and 30 minutes have past. Since the map is meant for ultra lowpop, I also would plan to add that to automatically start when players are less than <3 and the round starts, but fails for more than 3 times. In every attempt I try to make a new gamemode from this, it just fails for me.

## Why It's Good For The Game
RP and to do something for TGMC not to make sure on it's dead hours.

## Changelog
:cl:
add: Added the TGB - "Rising Star", an ultra lowpop map designed for RP, original map from Kwin and co. for ball. Only meant for Extended.
fix: Fixed rooms from the original map, added new rooms to compliment it.
add: Added a practice laser rifle for the said map, only deals stamina damage.
admin: Since "Rising Star" is meant for the off-duty gamemode, and it cannot be finished because I'm a noob to coding, it's best to switch the gamemode to Extended until some admin finishes the round manually.
/:cl:

### CODE REVIEW

**Comment by Alterist** [comment]
i'm gonna need you to update your branch, friendo

**Comment by XSlayer300** [comment]
> i'm gonna need you to update your branch, friendo

Done m8.

File: code/game/objects/machinery/floodlight.dm [hunk 1]
```diff
@@ -142,3 +142,7 @@
 	icon_state = "flood01"
 	on = TRUE
 	use_power = FALSE
+
+/obj/machinery/floodlight/landing/hq
+	name = "Installation Light"
+	desc = "A powerful light stationed on the base to provide better visibility."
```
Comment (spookydonut) [inline code/game/objects/machinery/floodlight.dm]:
```suggestion
	desc = "A powerful light stationed on the base to provide better visibility."

```

File: code/modules/projectiles/magazines/energy.dm [hunk 2]
```diff
@@ -52,5 +52,13 @@
...
+	name = "\improper M43-P lasgun battery"
+	desc = "A specialized high density battery used to power the M43 Lasgun. It lacks any potential to harm someone, but it has the ability to recharge."
+	gun_type = /obj/item/weapon/gun/energy/lasgun/M43/practice
+	self_recharge = TRUE
+	charge_amount = 25 // 10%, 2 shots
+	charge_delay = 2 SECONDS
+
 /obj/item/cell/lasgun/update_icon()
-	return FALSE
+	return FALSE
```
Comment (spookydonut) [inline code/modules/projectiles/magazines/energy.dm]:
```suggestion
	return FALSE

```

**Comment by XSlayer300** [comment]
Eh, this is going to be enough for a review by now.

**Comment by spookydonut** [comment]
file ./code/datums/gamemodes/offduty.dm is missing a trailing newline

file ./code/game/objects/machinery/floodlight.dm is missing a trailing newline

**Comment by spookydonut** [comment]
Also looks like a messy merge

**Comment by spookydonut** [comment]
Yep d2ff34b752c2a17042b06818ec4da3169045d460

**Comment by XSlayer300** [comment]
Yeahhhh... kind of my fault that it is messy, sorry.

File: code/modules/projectiles/ammo_datums.dm [hunk 3]
```diff
@@ -25,7 +25,7 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	var/scatter  					= 0 		// How much the ammo scatters when burst fired, added to gun scatter, along with other mods
 	var/damage 						= 0 		// This is the base damage of the bullet as it is fired
 	var/damage_falloff 				= 1 		// How much damage the bullet loses per turf traveled
-	var/damage_type 				= BRUTE 	// BRUTE, BURN, TOX, OXY, CLONE are the only things that should be in here
+	var/damage_type 				= BRUTE 	// BRUTE, BURN, TOX, OXY, CLONE, STAMINA are the only things that should be in here
```
Comment (spookydonut) [inline code/modules/projectiles/ammo_datums.dm]: why

File: code/modules/projectiles/guns/energy.dm [hunk 4]
```diff
@@ -386,7 +386,7 @@
 	cell_type = /obj/item/cell/lasgun/pulse
 	charge_cost = M43_STANDARD_AMMO_COST
 
-	flags_gun_features = GUN_AUTO_EJECTOR|GUN_CAN_POINTBLANK|GUN_AMMO_COUNTER|GUN_ENERGY|GUN_AMMO_COUNTER
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_CAN_POINTBLANK|GUN_AMMO_COUNTER|GUN_ENERGY
```
Comment (spookydonut) [inline code/modules/projectiles/guns/energy.dm]: why

**Comment by XSlayer300** [comment]
Both fixed.

File: code/game/objects/machinery/floodlight.dm [hunk 5]
```diff
@@ -18,7 +18,6 @@
 	. = ..()
 	cell = new /obj/item/cell(src)
 
-
```
Comment (spookydonut) [inline code/game/objects/machinery/floodlight.dm]: avoid these kind of changes in future

File: config/shipmaps.txt [hunk 6]
```diff
@@ -21,4 +21,9 @@ map theseus
 endmap
 
 map pillar_of_spring
+endmap
+
+map risingstar
+	voteweight 0
```
Comment (spookydonut) [inline config/shipmaps.txt]: these options arent used.

File: config/shipmaps.txt [hunk 7]
```diff
@@ -22,8 +22,3 @@ endmap
 
 map pillar_of_spring
 endmap
-
-map risingstar
```
Comment (spookydonut) [inline config/shipmaps.txt]: you still need this part

File: config/shipmaps.txt [hunk 8]
```diff
@@ -22,8 +22,3 @@ endmap
...
-	voteweight 0
-	maxplayers 5
-endmap
```
Comment (spookydonut) [inline config/shipmaps.txt]: and this part

Comment (XSlayer300) [reply config/shipmaps.txt]: Already done.

**Comment by spookydonut** [comment]
Job PR renamed the landmark paths, you'll need @Rohesie  to tell you which is which as that pr wasnt documented well

**Pull request merged**